### PR TITLE
plot_param_importances with target values other than objective value

### PR DESCRIPTION
**Pull request by not522**
## Motivation
This PR enables flexible plotting in `plot_param_importances`.
Related to #2064, #2109.
## Description of the changes
Added `target` argument to `plot_param_importances`.

### CODE REVIEW

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. 

I am concerned that `target` can be specified, but `target_name` cannot. Even if you switch the `target` and evaluate the importance and plot it, we cannot know from the figure what kind of target it is. 

For example, one solution would be to put the title `Importances for Objective Values` on the figure. I would like to hear other reviewers' opinions as well.

**Comment by not522** [comment]
I updated PR following your comments.

File: optuna/visualization/_param_importances.py [hunk 1]
```diff
@@ -81,6 +86,11 @@ def objective(trial):
             A list of names of parameters to assess.
             If :obj:`None`, all parameters that are present in all of the completed trials are
             assessed.
+        target:
+            A function to specify the value to evaluate importances. If it is :obj:`None`, the
+            objective values are used.
+        target_name:
+            Target's name to display on the axis label.
```
Comment (HideakiImamura) [inline optuna/visualization/_param_importances.py:93]: We set this name as not the axis label but the title.
```suggestion
            Target's name to display on the title. The title would be "Importance for ``target_name``".
```

File: optuna/visualization/matplotlib/_param_importances.py [hunk 2]
```diff
@@ -93,26 +98,33 @@ def objective(trial):
             A list of names of parameters to assess.
             If :obj:`None`, all parameters that are present in all of the completed trials are
             assessed.
+        target:
+            A function to specify the value to evaluate importances. If it is :obj:`None`, the
+            objective values are used.
+        target_name:
+            Target's name to display on the axis label.
```
Comment (HideakiImamura) [inline optuna/visualization/matplotlib/_param_importances.py:105]: Ditto.
```suggestion
            Target's name to display on the title. The title would be "Importance for ``target_name``".
```

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the update! I have two minor comments. Please take a look.

**Comment by HideakiImamura** [comment]
@toshihikoyanase Could you review this PR if you have time?

File: optuna/visualization/_param_importances.py (hunk 1 above)
Comment (not522) [reply optuna/visualization/_param_importances.py:93]: It's the title of the xaxis. I think "axis label" makes sense.

Comment (HideakiImamura) [reply optuna/visualization/_param_importances.py:93]: Ops! Sorry for the confusion. Let me resolve this conversation without any changes.

**Review by HideakiImamura — APPROVED** [review]
LGTM!

**Review by toshihikoyanase — APPROVED** [review]
Both methods worked as expected. LGTM!

plotly
image

matplotlib
tmp

**Pull request merged**